src/rpcq.lisp

```diff
@@ -559,22 +556,22 @@ We distinguish between the following options for any field type:
              (lambda (slot-name)
                `( ,(intern (symbol-name slot-name) :keyword)
                   (%deserialize (gethash ,(symbol-name slot-name) ,json))))))
-      (alexandria:with-gensyms (obj)
+      (alexandria:with-gensyms (obj hash-table)
         (let ((slot-names (mapcar #'car field-specs)))
           `(progn
              (defclass ,class-name ()
                ,(mapcar #'make-slot-spec field-specs)
                ,@(when documentation
                    `((:documentation ,(format-documentation-string documentation)))))
-
-             (defmethod yason:encode ((,obj ,class-name) &optional (stream *standard-output*))
+             
+             (defmethod %serialize ((,obj ,class-name))
                (with-slots ,slot-names ,obj
-                 (yason:with-output (stream)
-                   (yason:with-object ()
-                     (yason:encode-object-element "_type" ,(symbol-name class-name))
-                     ,@(loop :for slot :in slot-names
-                             :collect `(yason:encode-object-element ,(string-downcase (symbol-name slot))
-                                                                    ,slot))))))
+                 (let ((,hash-table (make-hash-table :test #'equal)))
+                   (setf (gethash "_type" ,hash-table) ,(symbol-name class-name))
+                   ,@(loop :for slot :in slot-names
+                           :collect `(setf (gethash ,(string-downcase (symbol-name slot)) ,hash-table)
```

Comment:
Do we mean to downcase if we are specifically writing `|slots|` like so?

Reply:
It looks like that has to be consistent with the `(gethash ,(symbol-name slot-name) ,json)` on line 566, which doesn't have a corresponding downcase. I think my preference is to put downcases in both places? but then it seems like the pipes are worthless.

Reply:
I'm going to drop the downcase here.